src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -652,9 +652,11 @@ private static Entry<DetailAST, Integer> calculateDistanceBetweenScopes(
                     default:
                         exprWithVariableUsage = blockWithVariableUsage.getFirstChild();
                 }
-                currentScopeAst = exprWithVariableUsage;
                 variableUsageAst =
                         Objects.requireNonNullElse(exprWithVariableUsage, blockWithVariableUsage);
+                if (exprWithVariableUsage != null) {
+                    currentScopeAst = variableUsageAst;
```

Comment:
The logic is hard to read / understand. Reading this line you need to keep in mind both the `if` condition and the condition of `Objects.requireNonNullElse`.

It is also strange to me that the mutated code hasn't been touched at all. You just changed unrelated code. Have you tried to hardcode mutation? What was the result?

Can you try to replace a call to `Objects.requireNonNullElse` and your new condition with the following?
```
                if (exprWithVariableUsage == null) {
                    variableUsageAst = blockWithVariableUsage;
                } else {
                    variableUsageAst = exprWithVariableUsage;
                    currentScopeAst = variableUsageAst;
                }
```

Reply:
> tried to hardcode mutation? What was the result?
Yes the report is clean https://github.com/checkstyle/checkstyle/pull/13121#issuecomment-1571651427

> Can you try to replace a call to Objects.requireNonNullElse and your new condition with the following?
@rdiachenko In the first try I have done the same as you but i have not include `currentScopeAst = variableUsageAst;`
in else thats why mutation has not killed at that time and I have make this type of changes.

I have updated the pr with the changes you suggest it will kill the mutation

Reply:
done